gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/jupiter/handlers/api/v4/DefaultApiReactor.java

```diff
@@ -313,6 +368,29 @@ private void setApiResponseTimeMetric(ExecutionContext ctx) {
         }
     }
 
+    private Completable timeout(final Completable upstream, MutableExecutionContext ctx) {
+        // When timeout is configured with 0 or less, consider it as infinity: no timeout operator to use in the chain.
+        if (httpRequestTimeoutConfiguration.getHttpRequestTimeout() <= 0) {
+            return upstream;
+        }
+
+        return Completable.defer(
+            () ->
+                upstream.timeout(
+                    Math.max(
+                        httpRequestTimeoutConfiguration.getHttpRequestTimeoutGraceDelay(),
```

Comment:
The reactor is about any kind of API not only HTTP based. Should we not use another kind of timeout here?

Reply:
It as been rename to `getRequestTimeoutGraceDelay` so I guess it's ok now and can be applied whatever the type of api.